medusa/server/api/v2/show.py

```diff
@@ -46,70 +55,147 @@ def get(self, show_id):
-                'next_ep_airdate': sbdatetime.sbdatetime.sbfdate(dt_episode_airs, d_preset=dateFormat) if dt_episode_airs else '',
+                'nextAirdate': sbdatetime.sbdatetime.sbfdate(dt_episode_airs, d_preset=dateFormat) if dt_episode_airs else '',
                 'status': show.status,
                 'subtitles': bool(show.subtitles),
-                'cache': CMD_ShowCache((), {'indexerid': show.indexerid}).run()['data']
+                'cache': {
+                    'poster': bool(os.path.isfile(cache_obj.poster_path(show.indexerid))),
+                    'banner': bool(os.path.isfile(cache_obj.banner_path(show.indexerid)))
+                },
+                'startYear': show.imdb_info['year'] if show.imdb_info.get('year') else show.startyear,
+                'runtime': show.imdb_info['runtimes'] if show.imdb_info.get('runtimes') else show.runtime,
+                'countries': show.imdb_info['country_codes'].split('|') if 'country_codes' in show.imdb_info else []
             }
 
+            if app.showQueueScheduler.action.isBeingAdded(show):
+                show_dict.queue.raw = 'isBeingAdded'
+                show_dict.queue.pretty = 'This show is in the process of being downloaded - the info below is incomplete.'
+
+            elif app.showQueueScheduler.action.isBeingUpdated(show):
+                show_dict.queue.raw = 'isBeingUpdated'
+                show_dict.queue.pretty = 'The information on this page is in the process of being updated.'
+
+            elif app.showQueueScheduler.action.isBeingRefreshed(show):
+                show_dict.queue.raw = 'isBeingRefreshed'
+                show_dict.queue.pretty = 'The episodes below are currently being refreshed from disk'
+
+            elif app.showQueueScheduler.action.isBeingSubtitled(show):
+                show_dict.queue.raw = 'isBeingSubtitled'
+                show_dict.queue.pretty = 'Currently downloading subtitles for this show'
+
+            elif app.showQueueScheduler.action.isInRefreshQueue(show):
+                show_dict.queue.raw = 'isInRefreshQueue'
+                show_dict.queue.pretty = 'This show is queued to be refreshed.'
+
+            elif app.showQueueScheduler.action.isInUpdateQueue(show):
+                show_dict.queue.raw = 'isInUpdateQueue'
+                show_dict.queue.pretty = 'This show is queued and awaiting an update.'
+
+            elif app.showQueueScheduler.action.isInSubtitleQueue(show):
+                show_dict.queue.raw = 'isInSubtitleQueue'
+                show_dict.queue.pretty = 'This show is queued and awaiting subtitles download.'
+
             # Detailed information
             if show_id:
                 any_qualities, best_qualities = _map_quality(show.quality)
 
-                # @TODO: Replace these with commands from here
+                # ------ START OF SEASONS ------
+                main_db_con = db.DBConnection(row_type='dict')
```

Comment:
@fernandog @ratoaq2 and I talked about it and get_all_episodes works on the show object but every episode needs to be in JSON so we really need a to_json method added to the TVEpisode class as well as the TVShow class if possible.